webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java

```diff
@@ -85,4 +93,88 @@
 
         return data;
 	}
+
+    @Resource
+    private SecurityService securityService = (SecurityService) ComponentManager.get(SecurityService.class);
+
+    @Resource
+    private PreferencesService preferencesService = (PreferencesService) ComponentManager.get(PreferencesService.class);
+
+    @Resource
+    private ActiveToolManager activeToolManager = (ActiveToolManager) ComponentManager.get(ActiveToolManager.class);
+
+    @Resource
+    private ToolManager toolManager = (ToolManager) ComponentManager.get(ToolManager.class.getName());
+
+	@GetMapping(value = "/users/{userId}/favs", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, List<Map<String, Object>>> getFavsWithPages(@PathVariable String userId)
+        throws UserNotDefinedException {
+
+        Session session = checkSakaiSession();
+        String reqUser = session.getUserId();
+        //If non admin user requests data of another user, return empty map and warn
+        if(!securityService.isSuperUser() && !userId.equals(reqUser)) {
+            log.warn("User " + reqUser + " requested favorites from user " + userId);
+            return new HashMap<>();
+        }
+        return getUserFavoriteSitesWithPages(userId);
+	}
+
+	@GetMapping(value = "/user/favs", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, List<Map<String, Object>>> getFavsWithPages()
+        throws UserNotDefinedException {
+
+        Session session = checkSakaiSession();
+        String reqUser = session.getUserId();
+        return getUserFavoriteSitesWithPages(reqUser);
+	}
+
+    private Map<String, List<Map<String, Object>>> getUserFavoriteSitesWithPages(String userId)
+        throws UserNotDefinedException  {
+
+        ResourceProperties props = preferencesService.getPreferences(userId).getProperties(org.sakaiproject.user.api.PreferencesService.SITENAV_PREFS_KEY);
+        List<Map<String, Object>> sites = props.getPropertyList("order").stream().map(f -> {
+            Map<String, Object> site = new HashMap<>();
+            try {
+                Site s = siteService.getSite(f);
+                site.put("id", s.getId());
+                site.put("title", s.getTitle());
+                site.put("url", s.getUrl());
+                site.put("type", s.getType());
```

Comment:
Is suspect we'll need more properties in the future like the term etc, enough for now?

Reply:
I'm not sure yet, what we will need later, I'll add it when I do.